graph/graph.go

```diff
@@ -18,6 +18,11 @@ type Edge interface {
 	Weight() float64
 }
 
+// WeightedEdge is a graph edge. In directed graphs, the direction
+// of the edge is given from -> to, otherwise the edge is semantically
+// unordered.
+type WeightedEdge Edge
```

Comment:
As already mentioned, later the `Weight` method should be added here and removed from `Edge`.